src/components/general/Dropdown/index.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useState, useRef, useCallback, FC, useEffect } from 'react';
+import {
+    TextInput,
+    DropdownArrow,
+    Menu,
+    useClickOutsideOverlayPortal,
+    RelativeOverlayPortal,
+} from '@equinor/fusion-components';
+import styles from './styles.less';
+
+type DropdownOption = {
+    title: string;
+    key: string;
+};
+
+type DropdownProps = {
+    label?: string;
+    options: DropdownOption[];
+    onSelect?: (item: DropdownOption) => void;
+    selected?: string;
+};
+
+const Dropdown: FC<DropdownProps> = ({ options, label, onSelect, selected }) => {
+    const [open, setOpen] = useState(false);
+    const [inputValue, setInputValue] = useState('');
+    const [dropdownOption, setDropdownOption] = useState<DropdownOption[]>([]);
+
+    const inputRef = useRef<HTMLInputElement | null>(null);
+    const inputContainerRef = useRef<HTMLDivElement | null>(null);
+
+    useEffect(() => setDropdownOption(options), [options])
+
+    const close = useCallback(() => open && setOpen(false), [open]);
+
+    const filterSearch = useCallback(inputValue => {
+        setDropdownOption(
+            options.filter(option =>
+                option.title.toLowerCase().includes(inputValue.toLowerCase())
+            )
+        );
+    }, [options]);
+
+    useClickOutsideOverlayPortal(close, inputContainerRef.current);
+
+    return (
+        <div className={styles.inputContainer} ref={inputContainerRef}>
+            <TextInput
+                onChange={value => {
+                    if (!open) {
+                        setOpen(true);
+                        return;
+                    }
+                    setInputValue(value);
+                    filterSearch(value);
```

Comment:
I guess the correct way to perform the filter would be in a useEffect with dependency on the input value? 🤔